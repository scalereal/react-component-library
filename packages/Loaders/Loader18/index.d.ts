import * as React from 'react';
import { WithStyle } from '../../utils';
import { LoaderProps } from './types';
declare const Loader18: React.FC<LoaderProps> & WithStyle;
export default Loader18;
