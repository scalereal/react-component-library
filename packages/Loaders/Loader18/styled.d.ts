import { LoaderProps } from './types';
export declare const Loader: import("styled-components").StyledComponent<"div", any, LoaderProps, never>;
export declare const LoaderWrapper: import("styled-components").StyledComponent<"div", any, LoaderProps, never>;
export declare const StyledWrapper: import("styled-components").StyledComponent<"div", any, LoaderProps, never>;
