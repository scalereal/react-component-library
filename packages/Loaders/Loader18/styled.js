var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
var mixin = function (size, Color) {
    switch (size) {
        case "XXS":
            return "\n                height: 15px;\n                width: 15px;\n                border: 1px solid transparent;\n                border-top: 1px solid " + (Color || '#FF5722') + ";\n                &:before{\n                    top: 2px;\n                    right: 2px;\n                    bottom: 2px;\n                    left: 2px;\n                    border: 1px solid transparent;\n                }\n                &:after{\n                    top: 5px;\n                    right: 5px;\n                    bottom: 5px;\n                    left: 5px;\n                    border: 1px solid transparent;\n                }\n           ";
        case "XS":
            return "\n                height: 20px;\n                width: 20px;\n                border: 2px solid transparent;\n                border-top: 2px solid " + (Color || '#FF5722') + ";\n                &:before{\n                    top: 2px;\n                    right: 2px;\n                    bottom: 2px;\n                    left: 2px;\n                    border: 2px solid transparent;\n                }\n                &:after{\n                    top: 5px;\n                    right: 5px;\n                    bottom: 5px;\n                    left: 5px;\n                    border: 2px solid transparent;\n                }\n            ";
        case "S":
            return "\n            height: 40px;\n            width: 40px;\n            border: 4px solid transparent;\n            border-top: 4px solid " + (Color || '#FF5722') + ";\n            &:before{\n                top: 3px;\n                right: 3px;\n                bottom: 3px;\n                left: 3px;\n                border: 4px solid transparent;\n            }\n            &:after{\n                top: 10px;\n                right: 10px;\n                bottom: 10px;\n                left: 10px;\n                border: 4px solid transparent;\n            }\n            ";
        case "M":
            return "\n            height: 60px;\n            width: 60px;\n            border: 5px solid transparent;\n            border-top: 5px solid " + (Color || '#FF5722') + ";\n            &:before{\n                top: 4px;\n                right: 4px;\n                bottom: 4px;\n                left: 4px;\n                border: 5px solid transparent;\n            }\n            &:after{\n                top: 15px;\n                right: 15px;\n                bottom: 15px;\n                left: 15px;\n                border: 5px solid transparent;\n            }\n            ";
        case "L":
            return "\n                border: 6px solid transparent;\n                height: 80px;\n                width: 80px;\n                border-top: 6px solid " + (Color || '#FF5722') + ";\n            &:before{\n                top: 5px;\n                right: 5px;\n                bottom: 5px;\n                left: 5px;\n                border: 6px solid transparent;\n            }\n            &:after{\n                top: 18px;\n                right: 18px;\n                bottom: 18px;\n                left: 18px;\n                border: 6px solid transparent;\n            }\n            ";
        case "XL":
            return "\n            height: 120px;\n            width: 120px;\n            border: 6px solid transparent;\n            border-top: 6px solid " + (Color || '#FF5722') + ";\n            &:before{\n                top: 7px;\n                right: 7px;\n                bottom: 7px;\n                left: 7px;\n                border: 6px solid transparent;\n            }\n            &:after{\n                top: 25px;\n                right: 25px;\n                bottom: 25px;\n                left: 25px;\n                border: 6px solid transparent;\n            }\n            ";
    }
};
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    ", "\n\n    display: block;\n    position: relative;\n    border-radius: 50%;\n    -webkit-animation: spin 2s linear infinite;\n    animation: spin 2s linear infinite;\n\n    &:before,\n    &:after {\n        content: \"\";\n        position: absolute;\n        border-radius: 50%;\n    }\n    &:before {\n        border-top-color:", ";\n        -webkit-animation: spin 3s linear infinite;\n        animation: spin 3.5s linear infinite;\n    }\n    &:after {\n        border-top-color: ", ";\n        -webkit-animation: spin 1.5s linear infinite;\n        animation: spin 1.75s linear infinite;\n    }\n\n    @-webkit-keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n    @keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n"], ["\n    ", "\n\n    display: block;\n    position: relative;\n    border-radius: 50%;\n    -webkit-animation: spin 2s linear infinite;\n    animation: spin 2s linear infinite;\n\n    &:before,\n    &:after {\n        content: \"\";\n        position: absolute;\n        border-radius: 50%;\n    }\n    &:before {\n        border-top-color:", ";\n        -webkit-animation: spin 3s linear infinite;\n        animation: spin 3.5s linear infinite;\n    }\n    &:after {\n        border-top-color: ", ";\n        -webkit-animation: spin 1.5s linear infinite;\n        animation: spin 1.75s linear infinite;\n    }\n\n    @-webkit-keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n    @keyframes spin {\n        from {\n            -webkit-transform: rotate(0deg);\n            transform: rotate(0deg);\n        }\n        to {\n            -webkit-transform: rotate(360deg);\n            transform: rotate(360deg);\n        }\n    }\n\n"])), function (pr) { return mixin(pr.size, pr.Color); }, function (pr) { return pr.innerColor || "#FF9800"; }, function (pr) { return pr.coreColor || "#FFC107"; });
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return pr.visible ? 'inline-block' : 'none'; });
export var StyledWrapper = styled.div(templateObject_3 || (templateObject_3 = __makeTemplateObject(["\n    -webkit-box-flex: 0;\n    -ms-flex: 0 0 25%;\n    flex: 0 0 25%;\n    border: 1px solid rgba(255, 255, 255, 0.1);\n    -webkit-box-sizing: border-box;\n    box-sizing: border-box;\n    margin: 0;\n    position: relative;\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n    -webkit-box-pack: center;\n    -ms-flex-pack: center;\n    justify-content: center;\n    -webkit-box-align: center;\n    -ms-flex-align: center;\n    align-items: center;\n    overflow: hidden;\n"], ["\n    -webkit-box-flex: 0;\n    -ms-flex: 0 0 25%;\n    flex: 0 0 25%;\n    border: 1px solid rgba(255, 255, 255, 0.1);\n    -webkit-box-sizing: border-box;\n    box-sizing: border-box;\n    margin: 0;\n    position: relative;\n    display: -webkit-box;\n    display: -ms-flexbox;\n    display: flex;\n    -webkit-box-pack: center;\n    -ms-flex-pack: center;\n    justify-content: center;\n    -webkit-box-align: center;\n    -ms-flex-align: center;\n    align-items: center;\n    overflow: hidden;\n"])));
var templateObject_1, templateObject_2, templateObject_3;
