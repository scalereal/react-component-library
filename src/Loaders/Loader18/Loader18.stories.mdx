import Loader18 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader18"
    component={Loader18}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        innerColor: {
            control: 'color'
        },
        coreColor: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader18 {...args} />
    </div>
);

<Preview>
    <Story name="Loader18" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader18 size="S" Color="green" innerColor="red" />
        <Loader18 size="S" Color="red" innerColor="green" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader18 size="XXS" />
        <Loader18 size="XS" />
        <Loader18 size="S" />
        <Loader18 size="M" />
        <Loader18 size="L" />
        <Loader18 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader18" />
