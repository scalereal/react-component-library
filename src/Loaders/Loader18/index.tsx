import React from 'react';
import { WithStyle } from '../../utils';
import { Loader, LoaderWrapper, StyledWrapper } from './styled';
import { LoaderProps } from './types';

const Loader18:React.FC<LoaderProps> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <StyledWrapper {...props}>
                <Loader {...props}/>
            </StyledWrapper>
        </LoaderWrapper>
    );
    
}));

Loader18.displayName="Loader18";
Loader18.defaultProps = {
    size:'S',
    visible: true
};

export default Loader18;
