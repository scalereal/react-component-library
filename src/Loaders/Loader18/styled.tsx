import styled from 'styled-components';
import { LoaderProps } from './types';

const mixin = (size:any,Color:any) => {
    switch (size) {
        case "XXS":
            return `
                height: 15px;
                width: 15px;
                border: 1px solid transparent;
                border-top: 1px solid ${Color || '#FF5722'};
                &:before{
                    top: 2px;
                    right: 2px;
                    bottom: 2px;
                    left: 2px;
                    border: 1px solid transparent;
                }
                &:after{
                    top: 5px;
                    right: 5px;
                    bottom: 5px;
                    left: 5px;
                    border: 1px solid transparent;
                }
           `;
        case "XS":
            return `
                height: 20px;
                width: 20px;
                border: 2px solid transparent;
                border-top: 2px solid ${Color || '#FF5722'};
                &:before{
                    top: 2px;
                    right: 2px;
                    bottom: 2px;
                    left: 2px;
                    border: 2px solid transparent;
                }
                &:after{
                    top: 5px;
                    right: 5px;
                    bottom: 5px;
                    left: 5px;
                    border: 2px solid transparent;
                }
            `;
        case "S":
            return `
            height: 40px;
            width: 40px;
            border: 4px solid transparent;
            border-top: 4px solid ${Color || '#FF5722'};
            &:before{
                top: 3px;
                right: 3px;
                bottom: 3px;
                left: 3px;
                border: 4px solid transparent;
            }
            &:after{
                top: 10px;
                right: 10px;
                bottom: 10px;
                left: 10px;
                border: 4px solid transparent;
            }
            `;
        
        case "M":
            return `
            height: 60px;
            width: 60px;
            border: 5px solid transparent;
            border-top: 5px solid ${Color || '#FF5722'};
            &:before{
                top: 4px;
                right: 4px;
                bottom: 4px;
                left: 4px;
                border: 5px solid transparent;
            }
            &:after{
                top: 15px;
                right: 15px;
                bottom: 15px;
                left: 15px;
                border: 5px solid transparent;
            }
            `;

        case "L":
            return `
                border: 6px solid transparent;
                height: 80px;
                width: 80px;
                border-top: 6px solid ${Color || '#FF5722'};
            &:before{
                top: 5px;
                right: 5px;
                bottom: 5px;
                left: 5px;
                border: 6px solid transparent;
            }
            &:after{
                top: 18px;
                right: 18px;
                bottom: 18px;
                left: 18px;
                border: 6px solid transparent;
            }
            `;
        
        case "XL":
            return `
            height: 120px;
            width: 120px;
            border: 6px solid transparent;
            border-top: 6px solid ${Color || '#FF5722'};
            &:before{
                top: 7px;
                right: 7px;
                bottom: 7px;
                left: 7px;
                border: 6px solid transparent;
            }
            &:after{
                top: 25px;
                right: 25px;
                bottom: 25px;
                left: 25px;
                border: 6px solid transparent;
            }
            `;
  
    }
 };

export const Loader = styled.div<LoaderProps>`
    ${pr=> mixin(pr.size,pr.Color)}

    display: block;
    position: relative;
    border-radius: 50%;
    -webkit-animation: spin 2s linear infinite;
    animation: spin 2s linear infinite;

    &:before,
    &:after {
        content: "";
        position: absolute;
        border-radius: 50%;
    }
    &:before {
        border-top-color:${pr=> pr.innerColor || `#FF9800` };
        -webkit-animation: spin 3s linear infinite;
        animation: spin 3.5s linear infinite;
    }
    &:after {
        border-top-color: ${pr=> pr.coreColor || `#FFC107` };
        -webkit-animation: spin 1.5s linear infinite;
        animation: spin 1.75s linear infinite;
    }

    @-webkit-keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

    @keyframes spin {
        from {
            -webkit-transform: rotate(0deg);
            transform: rotate(0deg);
        }
        to {
            -webkit-transform: rotate(360deg);
            transform: rotate(360deg);
        }
    }

`;

export const LoaderWrapper = styled.div<LoaderProps>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    display: ${pr=>pr.display?'inline-block':'none'};
`;

export const StyledWrapper = styled.div<LoaderProps>`
    -webkit-box-flex: 0;
    -ms-flex: 0 0 25%;
    flex: 0 0 25%;
    border: 1px solid rgba(255, 255, 255, 0.1);
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    margin: 0;
    position: relative;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    overflow: hidden;
`;
