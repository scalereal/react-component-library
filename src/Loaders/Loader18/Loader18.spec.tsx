import { render } from '@testing-library/react';
import * as React from 'react';
import Loader18 from '.';

describe('Loader18 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader18/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader18
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader18 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});