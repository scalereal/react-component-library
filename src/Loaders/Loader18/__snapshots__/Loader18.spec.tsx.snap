// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader18 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ llzSXf"
      />
    </div>
  </div>
</div>
`;

exports[`Loader18 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ lfIUKy"
      />
    </div>
  </div>
</div>
`;

exports[`Loader18 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz hdkynM"
  >
    <div
      class="sc-dlfnbm bIJBxZ"
    >
      <div
        class="sc-bdfBwQ bHQCfv"
      />
    </div>
  </div>
</div>
`;
